src/v3/v2converter.ts

```diff
@@ -0,0 +1,228 @@
+import type { ReturnRate } from '../rates'
+import { ONE_MINUTE } from './constants'
+import type {
+  EdgeAsset,
+  GetRatesParams,
+  IncomingGetRatesParams,
+  V2CurrencyCodeMap
+} from './types'
+
+interface V2Request {
+  currency_pair: string
+  date?: string
+}
+
+interface ParsedPair {
+  from: string
+  to: string
+  isFiatFrom: boolean
+  isFiatTo: boolean
+  date?: Date
+}
+
+/**
+ * Parse a V2 currency pair like "BTC_iso:USD" or "iso:EUR_ETH"
+ */
+const parseCurrencyPair = (currencyPair: string, date?: string): ParsedPair => {
+  const parts = currencyPair.split('_')
+  if (parts.length !== 2) {
+    throw new Error(`Invalid currency pair format: ${currencyPair}`)
+  }
+
+  const [from, to] = parts
+  const isFiatFrom = from.startsWith('iso:')
+  const isFiatTo = to.startsWith('iso:')
+
+  return {
+    from: isFiatFrom ? from.slice(4) : from,
+    to: isFiatTo ? to.slice(4) : to,
+    isFiatFrom,
+    isFiatTo,
+    date: date != null ? new Date(date) : undefined
+  }
+}
+
+/**
+ * Convert a currency code to an EdgeAsset using the currency code map
+ * If not found in the map, uses the uppercase currency code as pluginId with null tokenId
+ */
+const currencyCodeToAsset = (
+  currencyCode: string,
+  currencyCodeMaps: V2CurrencyCodeMap
+): EdgeAsset => {
+  const mapping = currencyCodeMaps[currencyCode]
+  if (mapping == null) {
+    // Fallback: use uppercase currency code as pluginId
+    // This will likely result in an error in the V3 API response for this asset
+    return {
+      pluginId: currencyCode.toUpperCase(),
+      tokenId: null
+    }
+  }
+
+  return {
+    pluginId: mapping.pluginId,
+    tokenId: mapping.tokenId
+  }
+}
+
+export const convertV2 = (
+  v2Requests: V2Request[],
+  currencyCodeMaps: V2CurrencyCodeMap
+): IncomingGetRatesParams => {
+  // Get the time rounded down to the nearest minute
+  const roundedNow = new Date(
+    Math.floor(new Date().getTime() / ONE_MINUTE) * ONE_MINUTE
+  )
+
+  // Parse all V2 requests
+  const parsedPairs = v2Requests.map(req =>
+    parseCurrencyPair(req.currency_pair, req.date)
+  )
+
+  // Group requests by their requirements
+  // Since V3 requires targetFiat to be "USD", we need to organize our requests accordingly
+  const cryptoAssetsMap = new Map<
+    string,
+    { asset: EdgeAsset; isoDate: Date | undefined; rate: number | undefined }
+  >()
+  const fiatCodesMap = new Map<
+    string,
+    { fiatCode: string; isoDate: Date | undefined; rate: number | undefined }
+  >()
+
+  // Helper to add crypto asset
+  const addCrypto = (currencyCode: string, date: Date): void => {
+    const asset = currencyCodeToAsset(currencyCode, currencyCodeMaps)
+    const key = `${asset.pluginId}_${
```

Comment:
Use toCryptoKey util

Reply:
I mentioned the wrong util. There's toDatedCryptoKey (and toDatedFiatKey) hiding in the src/v3/router file instead of utils. For the very issue you mentioned it puts the date in the front